framework/aster-frame/src/sync/spin.rs

```diff
@@ -67,17 +68,27 @@ impl<T: ?Sized> SpinLock<T> {
     pub fn lock(&self) -> SpinLockGuard<T> {
         let guard = disable_preempt();
         self.acquire_lock();
-        SpinLockGuard {
+        SpinLockGuard_ {
             lock: self,
             inner_guard: InnerGuard::PreemptGuard(guard),
         }
     }
 
+    /// The same as [`lock`] but using dynamic borrow rules.
+    pub fn lock_arc(self: &Arc<Self>) -> ArcSpinLockGuard<T> {
```

Comment:
I am not aware of this concept called dynamic borrow rules. I don't think this has been formally defined by Rust language or mentioned in the Rust book.

Instead of telling the users _how_ you are doing it, it is more important to tell the users _what_ you are offering. The _what_ part is the contract, the _how_ part is the implementation. For the users, the most important thing to know is that the returned guard owns an `Arc`. I think our doc should resemble the one by [parking_lot's lock_arc](https://docs.rs/parking_lot/latest/parking_lot/type.Mutex.html#method.lock_arc).

Revise all Rust docs of the methods that mention "dynamic borrow rules".